src/sensors/softfusion/softfusionsensor.h

```diff
@@ -71,32 +95,114 @@ class SoftFusionSensor : public Sensor {
 		return true;
 	}
 
+	float lastReadTemperature = 0;
+	float temperatureChangeRate = 0;
+
+	float lastTemperatureAverage;
+	float temperatureSum = 0;
+	float temperatureTimeSum = 0;
+
+	static constexpr float temperatureAveragingSeconds = 5;
+
+	float zroChangeOverTemperature = 0;
+
+	void handleTemperatureMeasurement(float temperature, float timeStep) {
+#if USE_NONBLOCKING_CALIBRATION
```

Comment:
No C define blocks in softfusion core please :<
Should be separated as different (replaceable) calibration implementation, or at least separated in less spaghetti way :<